src/freertos_drivers/esp32/Esp32BootloaderHal.hxx

```diff
@@ -399,13 +424,16 @@ uint16_t flash_complete(void)
         if (res != ESP_OK)
         {
             LOG_ERROR("[Bootloader] Failed to update the boot partition!");
+            return openlcb::Defs::ERROR_FIRMWARE_CSUM;
         }
     }
-    else if (res == ESP_ERR_OTA_VALIDATE_FAILED)
+    else
```

Comment:
consider moving this else above the then part by inverting the condition. Then the entire function should look like this:
```
res = (call api)
if (res != ESP_OK) {
  log error
  return error code
}
res = (call api)
if (res != ESP_OK) {
  log error
  return error code
}
```
which saves on indent and is much easier to read.

Reply:
Done, also removed calls to `ESP_ERROR_CHECK_WITHOUT_ABORT` which are redundant since the error cases already include this information.